src/main/mi.mc

```diff
@@ -79,18 +75,19 @@ if lti (length argv) 2 then usage () else
   then
     -- Yes, split into program arguments (after stand alone '--')
     let split = splitDashDash rest in
-    let argvp = partition (isPrefix eqc "--") split.first in
-    let options = parseOptions argvp.0 in
+    let res : ArgResult Options = parseOptions split.first in
+    let options : Options = res.options in
+    let files : [String] = res.strings in
     maybePrintHelp options;
     -- Invoke the selected command
-    cmd argvp.1 options (cons "mi" split.last)
+    cmd files options (cons "mi" split.last)
   else
-    -- No, not a well known command
-    -- Parse options as far as possible. Does user require help?
-    let split = splitOptionPrefix (tail argv) in
-    let options = parseOptions split.first in
-    maybePrintHelp options;
+    -- No, not a well known command.
+    -- Does user require help?
+    let wantHelp = optionIsSome (find (eqString "--help") (tail argv)) in
+    maybePrintHelp {options with printHelp = wantHelp};
```

Comment:
I guess this makes it impossible to pass `--help` to the evaluated program when using the short syntax. Is that preferable to what we had before?

Reply:
I believe that's what we had before as well?

Reply:
Or wait, you could pass it after the filename before: `mi foo.mc --help`. I should add a call here to `splitOptionPrefix` to get that behaviour, I guess.